### PR TITLE
Unifies the es_failure flow

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

Unifies the es_failure flow

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

The change is covered in unittests as it depends on external API to fail, which is difficult to fake.

### CODE REVIEW

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -406,30 +406,21 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
...
-			foreach ( $error_messages as $error_message ) {
-				// Default stat for errors is 'error'
-				$stat = '.error';
-				// If curl error 28(timeout), the stat should be 'timeout'
-				if ( $this->is_curl_timeout( $error_message ) ) {
-					$stat = '.timeout';
-				}
+		if ( is_wp_error( $response ) || $response_code >= 400 ) {
+
+			$this->ep_handle_failed_request( $response, $statsd_prefix );
```
Comment (brandon-m-skinner) [inline search/includes/classes/class-search.php:417]:
```suggestion
		if ( is_wp_error( $response ) || $response_code >= 400 ) {
			$this->ep_handle_failed_request( $response, $statsd_prefix );
```

File: search/includes/classes/class-search.php [hunk 2]
```diff
@@ -406,30 +406,21 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
...
```
Comment (brandon-m-skinner) [inline search/includes/classes/class-search.php:418]:
```suggestion
```

**Review by brandon-m-skinner — CHANGES_REQUESTED** [review]
Looks good, just some minor whitespace changes to keep the same style.

**Comment by pschoffer** [comment]
Sweet thanks for looking into it!

**Review by brandon-m-skinner — APPROVED** [review]

**Comment by pschoffer** [comment]
Stacks-r1259

**Pull request merged**